src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/Common/GuestConfigurationCmdletBase.cs

```diff
@@ -0,0 +1,376 @@
+            // var policyDefinitions = PolicyClient.PolicyDefinitions.List().ToList();
+            var policyDefinitionsForTheInitiative = new List<PolicyDefinition>();
+
+            foreach(var policyDefinitionIdInInitiative in policyDefinitionIdsInInitiative)
+            {
+                var _initiativeName = GetInitiativeNameFromId(policyDefinitionIdInInitiative);
+                var policyDef = PolicyClient.PolicyDefinitions.GetBuiltIn(_initiativeName);
+                if(policyDef != null)
+                {
+                    policyDefinitionsForTheInitiative.Add(policyDef);
+                }
+            }
+
+            var gcPolicyAssignmentsList = new List<GuestConfigurationPolicyAssignment>();
+
+            // Get all gcrp assignments for the initiative - for policy definitions  of category "Guest Configuration", effectType "AuditIfNotExists"   
+            foreach(var policyDef in policyDefinitionsForTheInitiative)
+            {
+                var policyRule = JObject.Parse(policyDef.PolicyRule.ToString());
+                var policyRuleDictionary = policyRule.ToObject<Dictionary<string, object>>();
+
+                var policyRuleThen = JObject.Parse(policyRuleDictionary["then"].ToString());
+                var policyRuleThenDictionary = policyRuleThen.ToObject<Dictionary<string, object>>();
+                var effectType = policyRuleThenDictionary["effect"].ToString();
+
+                if (Constants.AuditIfNotExists.ToLower() != effectType.ToLower())
+                {
+                    continue;
+                }
+
+                var policyRuleThenDetails = JObject.Parse(policyRuleThenDictionary["details"].ToString());
+                var policyRuleDetailsDictionary = policyRuleThenDetails.ToObject<Dictionary<string, object>>();
+
+                if(Constants.GuestConfigurationRPType.ToLower() != policyRuleDetailsDictionary["type"].ToString().ToLower())
+                {
+                    continue;
+                }
+
+                var guestConfigurationAssignmentNameInPolicy = policyRuleDetailsDictionary["name"].ToString();
+
+                if (!string.IsNullOrEmpty(guestConfigurationAssignmentNameInPolicy))
+                {
+                    // check if there is guest configuration assignment in GCRP with the same name as guestConfigurationAssignmentNameInPolicy
+                    try
+                    {
+                        var gcrpAssignment = GuestConfigurationClient.GuestConfigurationAssignments.Get(resourceGroupName, guestConfigurationAssignmentNameInPolicy, vmName);
+                        gcPolicyAssignmentsList.Add(new GuestConfigurationPolicyAssignment(gcrpAssignment, policyDef.DisplayName));
+                    }
+                    catch (Management.GuestConfiguration.Models.ErrorResponseException exception)
+                    {
+                        if (HttpStatusCode.NotFound.Equals(exception.Response.StatusCode))
```

Comment:
why not use a where clause, to keep from rewriting the exception stacktrace for exceptions you should catch